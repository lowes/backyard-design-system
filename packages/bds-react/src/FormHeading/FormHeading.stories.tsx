import React from 'react'
import { GridV3 as Grid, FormHeading } from '../'
import { ApiLink } from '../utils/storybook/ApiLink'
import { withKnobs, text } from '@storybook/addon-knobs'

export default { title: '@lowes-tech/bds-react/FormHeading', decorators: [withKnobs] }

export const DefaultFormHeading = () => (
    <Grid.Column start={6} end={12}>
        <FormHeading>
            {text('Text', 'text')}
        </FormHeading>

        <ApiLink to='#' />
    </Grid.Column>
)