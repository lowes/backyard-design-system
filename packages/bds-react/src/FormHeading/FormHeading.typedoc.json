{
	"id": 0,
	"name": "@lowes-tech/bds-react/FormHeading",
	"kind": 1,
	"kindString": "Project",
	"flags": {},
	"children": [
		{
			"id": 15,
			"name": "default",
			"kind": 16777216,
			"kindString": "Reference",
			"flags": {},
			"target": 1
		},
		{
			"id": 2,
			"name": "FormHeadingProps",
			"kind": 256,
			"kindString": "Interface",
			"flags": {},
			"children": [
				{
					"id": 6,
					"name": "align",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"comment": {
						"shortText": "Aligns text to one of left, center, or right."
					},
					"type": {
						"type": "union",
						"types": [
							{
								"type": "literal",
								"value": "left"
							},
							{
								"type": "literal",
								"value": "center"
							},
							{
								"type": "literal",
								"value": "right"
							}
						]
					},
					"inheritedFrom": {
						"type": "reference",
						"name": "TypographyProps.align"
					}
				},
				{
					"id": 13,
					"name": "as",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"type": {
						"type": "reference",
						"typeArguments": [
							{
								"type": "intrinsic",
								"name": "any"
							}
						],
						"name": "ElementType"
					},
					"inheritedFrom": {
						"type": "reference",
						"name": "TypographyProps.as"
					}
				},
				{
					"id": 7,
					"name": "bold",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"comment": {
						"shortText": "Makes text bold weight."
					},
					"type": {
						"type": "intrinsic",
						"name": "boolean"
					},
					"inheritedFrom": {
						"type": "reference",
						"name": "TypographyProps.bold"
					}
				},
				{
					"id": 3,
					"name": "children",
					"kind": 1024,
					"kindString": "Property",
					"flags": {},
					"comment": {
						"shortText": "Text to use as heading"
					},
					"type": {
						"type": "reference",
						"name": "ReactNode"
					},
					"overwrites": {
						"type": "reference",
						"name": "TypographyProps.children"
					}
				},
				{
					"id": 4,
					"name": "className",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"comment": {
						"shortText": "DOM class name"
					},
					"type": {
						"type": "intrinsic",
						"name": "string"
					},
					"overwrites": {
						"type": "reference",
						"name": "TypographyProps.className"
					}
				},
				{
					"id": 9,
					"name": "color",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"comment": {
						"shortText": "Backyard Color Token\nExample: \"interactive\" or \"primary\""
					},
					"type": {
						"type": "union",
						"types": [
							{
								"type": "literal",
								"value": "marketing_interactive"
							},
							{
								"type": "literal",
								"value": "text_interactive"
							},
							{
								"type": "literal",
								"value": "text_interactive_inverse"
							},
							{
								"type": "literal",
								"value": "icon_interactive"
							},
							{
								"type": "literal",
								"value": "icon_interactive_inverse"
							},
							{
								"type": "literal",
								"value": "border_interactive"
							},
							{
								"type": "literal",
								"value": "border_interactive_inverse"
							},
							{
								"type": "literal",
								"value": "focus_interactive"
							},
							{
								"type": "literal",
								"value": "action_interactive"
							},
							{
								"type": "literal",
								"value": "action_interactive_hover"
							},
							{
								"type": "literal",
								"value": "action_interactive_pressed"
							},
							{
								"type": "literal",
								"value": "action_interactive_subdued"
							},
							{
								"type": "literal",
								"value": "action_interactive_subdued_hover"
							},
							{
								"type": "literal",
								"value": "action_interactive_subdued_pressed"
							},
							{
								"type": "literal",
								"value": "highlight"
							},
							{
								"type": "literal",
								"value": "marketing_dark_blue"
							},
							{
								"type": "literal",
								"value": "surface_dark_blue"
							},
							{
								"type": "literal",
								"value": "surface_dark_blue_subdued"
							},
							{
								"type": "literal",
								"value": "surface_dark_blue_inverse"
							},
							{
								"type": "literal",
								"value": "surface_dark_blue_subdued_inverse"
							},
							{
								"type": "literal",
								"value": "marketing_blue"
							},
							{
								"type": "literal",
								"value": "surface_blue"
							},
							{
								"type": "literal",
								"value": "surface_blue_subdued"
							},
							{
								"type": "literal",
								"value": "surface_blue_inverse"
							},
							{
								"type": "literal",
								"value": "surface_blue_subdued_inverse"
							},
							{
								"type": "literal",
								"value": "marketing_light_blue"
							},
							{
								"type": "literal",
								"value": "surface_light_blue"
							},
							{
								"type": "literal",
								"value": "surface_light_blue_subdued"
							},
							{
								"type": "literal",
								"value": "surface_light_blue_inverse"
							},
							{
								"type": "literal",
								"value": "surface_light_blue_subdued_inverse"
							},
							{
								"type": "literal",
								"value": "marketing_green"
							},
							{
								"type": "literal",
								"value": "text_green"
							},
							{
								"type": "literal",
								"value": "text_green_inverse"
							},
							{
								"type": "literal",
								"value": "icon_green"
							},
							{
								"type": "literal",
								"value": "icon_green_inverse"
							},
							{
								"type": "literal",
								"value": "border_green"
							},
							{
								"type": "literal",
								"value": "border_green_inverse"
							},
							{
								"type": "literal",
								"value": "surface_green"
							},
							{
								"type": "literal",
								"value": "surface_green_subdued"
							},
							{
								"type": "literal",
								"value": "surface_green_inverse"
							},
							{
								"type": "literal",
								"value": "surface_green_subdued_inverse"
							},
							{
								"type": "literal",
								"value": "action_green"
							},
							{
								"type": "literal",
								"value": "action_green_hover"
							},
							{
								"type": "literal",
								"value": "action_green_pressed"
							},
							{
								"type": "literal",
								"value": "action_green_subdued"
							},
							{
								"type": "literal",
								"value": "action_green_subdued_hover"
							},
							{
								"type": "literal",
								"value": "action_green_subdued_pressed"
							},
							{
								"type": "literal",
								"value": "marketing_red"
							},
							{
								"type": "literal",
								"value": "text_red"
							},
							{
								"type": "literal",
								"value": "text_red_inverse"
							},
							{
								"type": "literal",
								"value": "icon_red"
							},
							{
								"type": "literal",
								"value": "icon_red_inverse"
							},
							{
								"type": "literal",
								"value": "border_red"
							},
							{
								"type": "literal",
								"value": "border_red_inverse"
							},
							{
								"type": "literal",
								"value": "surface_red"
							},
							{
								"type": "literal",
								"value": "surface_red_subdued"
							},
							{
								"type": "literal",
								"value": "surface_red_inverse"
							},
							{
								"type": "literal",
								"value": "surface_red_subdued_inverse"
							},
							{
								"type": "literal",
								"value": "action_red"
							},
							{
								"type": "literal",
								"value": "action_red_hover"
							},
							{
								"type": "literal",
								"value": "action_red_pressed"
							},
							{
								"type": "literal",
								"value": "action_red_subdued"
							},
							{
								"type": "literal",
								"value": "action_red_subdued_hover"
							},
							{
								"type": "literal",
								"value": "action_red_subdued_pressed"
							},
							{
								"type": "literal",
								"value": "marketing_gold"
							},
							{
								"type": "literal",
								"value": "surface_gold"
							},
							{
								"type": "literal",
								"value": "surface_gold_subdued"
							},
							{
								"type": "literal",
								"value": "surface_gold_inverse"
							},
							{
								"type": "literal",
								"value": "surface_gold_subdued_inverse"
							},
							{
								"type": "literal",
								"value": "marketing_lfp_yellow"
							},
							{
								"type": "literal",
								"value": "text_solid_inverse"
							},
							{
								"type": "literal",
								"value": "text_primary_inverse"
							},
							{
								"type": "literal",
								"value": "text_secondary_inverse"
							},
							{
								"type": "literal",
								"value": "text_tertiary_inverse"
							},
							{
								"type": "literal",
								"value": "text_disabled_inverse"
							},
							{
								"type": "literal",
								"value": "icon_solid_inverse"
							},
							{
								"type": "literal",
								"value": "icon_primary_inverse"
							},
							{
								"type": "literal",
								"value": "icon_secondary_inverse"
							},
							{
								"type": "literal",
								"value": "icon_tertiary_inverse"
							},
							{
								"type": "literal",
								"value": "icon_disabled_inverse"
							},
							{
								"type": "literal",
								"value": "border_default_inverse"
							},
							{
								"type": "literal",
								"value": "border_hover_inverse"
							},
							{
								"type": "literal",
								"value": "border_pressed_inverse"
							},
							{
								"type": "literal",
								"value": "border_subdued_inverse"
							},
							{
								"type": "literal",
								"value": "border_contrast_inverse"
							},
							{
								"type": "literal",
								"value": "border_disabled_inverse"
							},
							{
								"type": "literal",
								"value": "focus_inverse"
							},
							{
								"type": "literal",
								"value": "surface_default"
							},
							{
								"type": "literal",
								"value": "surface_default_inverse"
							},
							{
								"type": "literal",
								"value": "surface_subdued"
							},
							{
								"type": "literal",
								"value": "elevation_01"
							},
							{
								"type": "literal",
								"value": "elevation_02"
							},
							{
								"type": "literal",
								"value": "elevation_03"
							},
							{
								"type": "literal",
								"value": "elevation_04"
							},
							{
								"type": "literal",
								"value": "elevation_05"
							},
							{
								"type": "literal",
								"value": "elevation_06"
							},
							{
								"type": "literal",
								"value": "action_neutral"
							},
							{
								"type": "literal",
								"value": "action_neutral_hover"
							},
							{
								"type": "literal",
								"value": "action_neutral_pressed"
							},
							{
								"type": "literal",
								"value": "white"
							},
							{
								"type": "literal",
								"value": "black"
							},
							{
								"type": "literal",
								"value": "text_solid"
							},
							{
								"type": "literal",
								"value": "text_primary"
							},
							{
								"type": "literal",
								"value": "text_secondary"
							},
							{
								"type": "literal",
								"value": "text_tertiary"
							},
							{
								"type": "literal",
								"value": "text_disabled"
							},
							{
								"type": "literal",
								"value": "icon_solid"
							},
							{
								"type": "literal",
								"value": "icon_primary"
							},
							{
								"type": "literal",
								"value": "icon_secondary"
							},
							{
								"type": "literal",
								"value": "icon_tertiary"
							},
							{
								"type": "literal",
								"value": "icon_disabled"
							},
							{
								"type": "literal",
								"value": "border_default"
							},
							{
								"type": "literal",
								"value": "border_hover"
							},
							{
								"type": "literal",
								"value": "border_pressed"
							},
							{
								"type": "literal",
								"value": "border_subdued"
							},
							{
								"type": "literal",
								"value": "border_contrast"
							},
							{
								"type": "literal",
								"value": "border_disabled"
							},
							{
								"type": "literal",
								"value": "focus_default"
							},
							{
								"type": "literal",
								"value": "neutral_01"
							},
							{
								"type": "literal",
								"value": "neutral_02"
							},
							{
								"type": "literal",
								"value": "neutral_03"
							},
							{
								"type": "literal",
								"value": "neutral_04"
							},
							{
								"type": "literal",
								"value": "neutral_05"
							},
							{
								"type": "literal",
								"value": "neutral_06"
							},
							{
								"type": "literal",
								"value": "neutral_07"
							},
							{
								"type": "literal",
								"value": "neutral_08"
							},
							{
								"type": "literal",
								"value": "neutral_09"
							},
							{
								"type": "literal",
								"value": "neutral_10"
							},
							{
								"type": "literal",
								"value": "skeleton_01"
							},
							{
								"type": "literal",
								"value": "skeleton_02"
							},
							{
								"type": "literal",
								"value": "overlay"
							},
							{
								"type": "literal",
								"value": "text_visited_link"
							}
						]
					},
					"inheritedFrom": {
						"type": "reference",
						"name": "TypographyProps.color"
					}
				},
				{
					"id": 10,
					"name": "component",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"comment": {
						"shortText": "Custom definition for HTML to use with Typography\nExample: \"p\" for paragraph or \"div\" for div"
					},
					"type": {
						"type": "reference",
						"typeArguments": [
							{
								"type": "intrinsic",
								"name": "any"
							}
						],
						"name": "ElementType"
					},
					"inheritedFrom": {
						"type": "reference",
						"name": "TypographyProps.component"
					}
				},
				{
					"id": 11,
					"name": "display",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"comment": {
						"shortText": "CSS Display type of inline or block"
					},
					"type": {
						"type": "union",
						"types": [
							{
								"type": "literal",
								"value": "block"
							},
							{
								"type": "literal",
								"value": "inline"
							}
						]
					},
					"inheritedFrom": {
						"type": "reference",
						"name": "TypographyProps.display"
					}
				},
				{
					"id": 12,
					"name": "marginBottom",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"comment": {
						"shortText": "Backyard spacing token for margin bottom.\nUseful for when you have a lot of headers and don't want to\nrecreate styles for each variant of header you are using."
					},
					"type": {
						"type": "union",
						"types": [
							{
								"type": "literal",
								"value": "size_0"
							},
							{
								"type": "literal",
								"value": "size_1"
							},
							{
								"type": "literal",
								"value": "size_2"
							},
							{
								"type": "literal",
								"value": "size_3"
							},
							{
								"type": "literal",
								"value": "size_4"
							},
							{
								"type": "literal",
								"value": "size_5"
							},
							{
								"type": "literal",
								"value": "size_6"
							},
							{
								"type": "literal",
								"value": "size_7"
							},
							{
								"type": "literal",
								"value": "size_8"
							},
							{
								"type": "literal",
								"value": "size_10"
							},
							{
								"type": "literal",
								"value": "size_12"
							},
							{
								"type": "literal",
								"value": "size_14"
							},
							{
								"type": "literal",
								"value": "size_16"
							},
							{
								"type": "literal",
								"value": "size_18"
							},
							{
								"type": "literal",
								"value": "size_20"
							},
							{
								"type": "literal",
								"value": "size_24"
							},
							{
								"type": "literal",
								"value": "size_28"
							},
							{
								"type": "literal",
								"value": "size_32"
							},
							{
								"type": "literal",
								"value": "size_36"
							},
							{
								"type": "literal",
								"value": "size_40"
							},
							{
								"type": "literal",
								"value": "size_44"
							},
							{
								"type": "literal",
								"value": "size_48"
							},
							{
								"type": "literal",
								"value": "size_52"
							},
							{
								"type": "literal",
								"value": "size_56"
							},
							{
								"type": "literal",
								"value": "size_60"
							},
							{
								"type": "literal",
								"value": "size_64"
							},
							{
								"type": "literal",
								"value": "size_68"
							},
							{
								"type": "literal",
								"value": "size_72"
							},
							{
								"type": "literal",
								"value": "size_76"
							},
							{
								"type": "literal",
								"value": "size_80"
							},
							{
								"type": "literal",
								"value": "size_84"
							},
							{
								"type": "literal",
								"value": "size_88"
							},
							{
								"type": "literal",
								"value": "size_92"
							},
							{
								"type": "literal",
								"value": "size_96"
							},
							{
								"type": "literal",
								"value": "size_100"
							},
							{
								"type": "literal",
								"value": "size_104"
							},
							{
								"type": "literal",
								"value": "size_108"
							},
							{
								"type": "literal",
								"value": "size_112"
							},
							{
								"type": "literal",
								"value": "size_116"
							},
							{
								"type": "literal",
								"value": "size_120"
							},
							{
								"type": "literal",
								"value": "size_124"
							},
							{
								"type": "literal",
								"value": "size_128"
							}
						]
					},
					"inheritedFrom": {
						"type": "reference",
						"name": "TypographyProps.marginBottom"
					}
				},
				{
					"id": 8,
					"name": "regular",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"comment": {
						"shortText": "Makes text regular weight."
					},
					"type": {
						"type": "intrinsic",
						"name": "boolean"
					},
					"inheritedFrom": {
						"type": "reference",
						"name": "TypographyProps.regular"
					}
				},
				{
					"id": 5,
					"name": "variant",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"comment": {
						"shortText": "Typography variant of heading\nNormally should use default value of 'h5'"
					},
					"type": {
						"type": "union",
						"types": [
							{
								"type": "literal",
								"value": "label"
							},
							{
								"type": "literal",
								"value": "h1"
							},
							{
								"type": "literal",
								"value": "h2"
							},
							{
								"type": "literal",
								"value": "h3"
							},
							{
								"type": "literal",
								"value": "h4"
							},
							{
								"type": "literal",
								"value": "h5"
							},
							{
								"type": "literal",
								"value": "h6"
							},
							{
								"type": "literal",
								"value": "body_1"
							},
							{
								"type": "literal",
								"value": "body_2"
							},
							{
								"type": "literal",
								"value": "article"
							},
							{
								"type": "literal",
								"value": "caption"
							},
							{
								"type": "literal",
								"value": "overline"
							},
							{
								"type": "literal",
								"value": "footnote"
							},
							{
								"type": "literal",
								"value": "helper"
							}
						]
					},
					"overwrites": {
						"type": "reference",
						"name": "TypographyProps.variant"
					}
				}
			],
			"groups": [
				{
					"title": "Properties",
					"kind": 1024,
					"children": [
						6,
						13,
						7,
						3,
						4,
						9,
						10,
						11,
						12,
						8,
						5
					]
				}
			],
			"extendedTypes": [
				{
					"type": "reference",
					"name": "TypographyProps"
				}
			]
		},
		{
			"id": 14,
			"name": "FormHeadingRef",
			"kind": 4194304,
			"kindString": "Type alias",
			"flags": {},
			"type": {
				"type": "reference",
				"name": "TypographyRef"
			}
		},
		{
			"id": 1,
			"name": "FormHeading",
			"kind": 32,
			"kindString": "Variable",
			"flags": {},
			"comment": {
				"shortText": "Backyard React Form Heading",
				"text": "`FormHeading` is intended to be used in conjunction with `FormControl`\n\nUse it on top of multiple form inputs inside `FormGroup` to give the\nuser visual context of the group of inputs. Example:\n\n```\n <FormControl>\n     <FormHeading>Heading Above FormGroup</FormHeading>\n     <FormGroup>\n         <Checkbox id=\"1\" />\n         <Checkbox id=\"2\" />\n         <Checkbox id=\"3\" />\n     </FormGroup>\n     <FormHelperText>Helper Text Below FormGroup</FormHelperText>\n </FormControl>\n```\n"
			},
			"type": {
				"type": "reference",
				"typeArguments": [
					{
						"type": "reference",
						"id": 2,
						"name": "FormHeadingProps"
					}
				],
				"name": "BDSForwardRef"
			},
			"defaultValue": "..."
		}
	],
	"groups": [
		{
			"title": "References",
			"kind": 16777216,
			"children": [
				15
			]
		},
		{
			"title": "Interfaces",
			"kind": 256,
			"children": [
				2
			]
		},
		{
			"title": "Type aliases",
			"kind": 4194304,
			"children": [
				14
			]
		},
		{
			"title": "Variables",
			"kind": 32,
			"children": [
				1
			]
		}
	]
}