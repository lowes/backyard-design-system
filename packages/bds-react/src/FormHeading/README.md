# @lowes-tech/bds-react/FormHeading

## Interfaces

- [FormHeadingProps](interfaces/FormHeadingProps.md)

## References

### default

Renames and re-exports [FormHeading](README.md#formheading)

## Type aliases

### FormHeadingRef

Ƭ **FormHeadingRef**: `TypographyRef`

## Variables

### FormHeading

• **FormHeading**: `BDSForwardRef`<[`FormHeadingProps`](interfaces/FormHeadingProps.md)\>

Backyard React Form Heading

`FormHeading` is intended to be used in conjunction with `FormControl`

Use it on top of multiple form inputs inside `FormGroup` to give the
user visual context of the group of inputs. Example:

```
 <FormControl>
     <FormHeading>Heading Above FormGroup</FormHeading>
     <FormGroup>
         <Checkbox id="1" />
         <Checkbox id="2" />
         <Checkbox id="3" />
     </FormGroup>
     <FormHelperText>Helper Text Below FormGroup</FormHelperText>
 </FormControl>
```
