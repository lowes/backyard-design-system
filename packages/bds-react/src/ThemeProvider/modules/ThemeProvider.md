# Namespace: ThemeProvider

## Variables

### bdsName

â€¢ **bdsName**: `string`
