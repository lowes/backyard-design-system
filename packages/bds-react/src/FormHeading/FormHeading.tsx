import * as React from 'react'
import classNames from 'classnames'

import Typography from '../Typography'
import type { TypographyRef, TypographyProps } from '../Typography'
import type { BDSForwardRef } from '../utils/typings/BDSComponentProps'
import { useBackyardTheme, validateBackyardTheme } from '../ThemeProvider'

/**
 * Backyard React Form Heading
 *
 * `FormHeading` is intended to be used in conjunction with `FormControl`
 *
 * Use it on top of multiple form inputs inside `FormGroup` to give the
 * user visual context of the group of inputs. Example:
 *
 * ```
 *  <FormControl>
 *      <FormHeading>Heading Above FormGroup</FormHeading>
 *      <FormGroup>
 *          <Checkbox id="1" />
 *          <Checkbox id="2" />
 *          <Checkbox id="3" />
 *      </FormGroup>
 *      <FormHelperText>Helper Text Below FormGroup</FormHelperText>
 *  </FormControl>
 * ```
 */
const FormHeading: BDSForwardRef<FormHeadingProps> = React.forwardRef<
    FormHeadingRef,
    FormHeadingProps
>(function FormHeading({ children, className, variant = 'h5', ...other }, ref) {
    // Get Backyard Theme Context
    const theme = useBackyardTheme()

    validateBackyardTheme(theme, 'FormHeading')

    /**
     * Layout:
     *
     *  <Typography>
     *      ...
     *  </Typography>
     */
    return (
        <Typography
            className={classNames('form-heading', className)}
            variant={variant}
            ref={ref as any}
            {...other}
        >
            {children}
        </Typography>
    )
})

type FormHeadingRef = TypographyRef

interface FormHeadingProps extends TypographyProps {
    /**
     * Text to use as heading
     */
    children: React.ReactNode
    /**
     * DOM class name
     */
    className?: string
    /**
     * Typography variant of heading
     * Normally should use default value of 'h5'
     */
    variant?: TypographyProps['variant']
}

FormHeading.bdsName = 'FormHeading'

export { FormHeading }

export type { FormHeadingProps, FormHeadingRef }

export default FormHeading
