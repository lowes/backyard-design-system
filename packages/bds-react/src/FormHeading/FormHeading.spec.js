import * as React from 'react'
import { render } from '../../test-utils'
import { ThemeProvider } from '../ThemeProvider'
import FormHeading from './FormHeading'

describe('FormHeading Snapshots', () => {
    test('FormHeading default snapshot', () => {
        const { asFragment } = render(
            <FormHeading>Heading</FormHeading>
        )

        expect(asFragment()).toMatchSnapshot()
    })

    describe('themes', () => {
        const themes = ['light', 'dark']

        themes.forEach((theme) => {
            test(`FormHeading ${theme} theme snapshot`, () => {
                const { asFragment } = render(
                    <ThemeProvider theme={theme}>
                        <FormHeading>Heading</FormHeading>
                    </ThemeProvider>
                )

                expect(asFragment()).toMatchSnapshot()
            })
        })
    })

    describe('prop: variant', () => {
        const variants = [undefined, 'caption', 'h5']

        variants.forEach((variant) => {
            test(`FormHeading ${variant ? variant : 'default'} snapshot`, () => {
                const { asFragment } = render(
                    <FormHeading variant={variant}>Heading</FormHeading>
                )

                expect(asFragment()).toMatchSnapshot()
            })
        })
    })
})
