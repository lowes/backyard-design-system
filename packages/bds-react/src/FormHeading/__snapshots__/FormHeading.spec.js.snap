// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormHeading Snapshots FormHeading default snapshot 1`] = `
<DocumentFragment>
  .c0 {
  color: hsl(0,0%,0%,.87);
  margin: 0;
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
  color: hsl(0,0%,0%,.87);
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  font-size: var(--bds-font-tag-h5-mobile-font-size);
  line-height: var(--bds-font-tag-h5-mobile-line-height);
}

.c0.align--left {
  text-align: left;
}

.c0.align--center {
  text-align: center;
}

.c0.align--right {
  text-align: right;
}

.c0.display--block {
  display: block;
}

.c0.display--inline {
  display: inline;
}

.c0 b,
.c0 bold,
.c0 strong {
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
}

.c0.bold {
  font-weight: 500;
}

@media only screen and (min-width:66.0625rem) {
  .c0 {
    font-size: var(--bds-font-tag-h5-desktop-font-size);
    line-height: var(--bds-font-tag-h5-desktop-line-height);
  }
}

<h5
    class="c0 typography variant--h5 align--left form-heading"
    color="text_primary"
  >
    Heading
  </h5>
</DocumentFragment>
`;

exports[`FormHeading Snapshots prop: variant FormHeading caption snapshot 1`] = `
<DocumentFragment>
  .c0 {
  color: hsl(0,0%,0%,.87);
  margin: 0;
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: var(--bds-font-weight-regular);
  color: hsl(0,0%,0%,.87);
  font-size: var(--bds-font-tag-caption-mobile-font-size);
  line-height: var(--bds-font-tag-caption-mobile-line-height);
}

.c0.align--left {
  text-align: left;
}

.c0.align--center {
  text-align: center;
}

.c0.align--right {
  text-align: right;
}

.c0.display--block {
  display: block;
}

.c0.display--inline {
  display: inline;
}

.c0 b,
.c0 bold,
.c0 strong {
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
}

.c0.bold {
  font-weight: 500;
}

@media only screen and (min-width:66.0625rem) {
  .c0 {
    font-size: var(--bds-font-tag-caption-desktop-font-size);
    line-height: var(--bds-font-tag-caption-desktop-line-height);
  }
}

<span
    class="c0 typography variant--caption align--left form-heading"
    color="text_primary"
  >
    Heading
  </span>
</DocumentFragment>
`;

exports[`FormHeading Snapshots prop: variant FormHeading default snapshot 1`] = `
<DocumentFragment>
  .c0 {
  color: hsl(0,0%,0%,.87);
  margin: 0;
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
  color: hsl(0,0%,0%,.87);
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  font-size: var(--bds-font-tag-h5-mobile-font-size);
  line-height: var(--bds-font-tag-h5-mobile-line-height);
}

.c0.align--left {
  text-align: left;
}

.c0.align--center {
  text-align: center;
}

.c0.align--right {
  text-align: right;
}

.c0.display--block {
  display: block;
}

.c0.display--inline {
  display: inline;
}

.c0 b,
.c0 bold,
.c0 strong {
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
}

.c0.bold {
  font-weight: 500;
}

@media only screen and (min-width:66.0625rem) {
  .c0 {
    font-size: var(--bds-font-tag-h5-desktop-font-size);
    line-height: var(--bds-font-tag-h5-desktop-line-height);
  }
}

<h5
    class="c0 typography variant--h5 align--left form-heading"
    color="text_primary"
  >
    Heading
  </h5>
</DocumentFragment>
`;

exports[`FormHeading Snapshots prop: variant FormHeading h5 snapshot 1`] = `
<DocumentFragment>
  .c0 {
  color: hsl(0,0%,0%,.87);
  margin: 0;
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
  color: hsl(0,0%,0%,.87);
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  font-size: var(--bds-font-tag-h5-mobile-font-size);
  line-height: var(--bds-font-tag-h5-mobile-line-height);
}

.c0.align--left {
  text-align: left;
}

.c0.align--center {
  text-align: center;
}

.c0.align--right {
  text-align: right;
}

.c0.display--block {
  display: block;
}

.c0.display--inline {
  display: inline;
}

.c0 b,
.c0 bold,
.c0 strong {
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
}

.c0.bold {
  font-weight: 500;
}

@media only screen and (min-width:66.0625rem) {
  .c0 {
    font-size: var(--bds-font-tag-h5-desktop-font-size);
    line-height: var(--bds-font-tag-h5-desktop-line-height);
  }
}

<h5
    class="c0 typography variant--h5 align--left form-heading"
    color="text_primary"
  >
    Heading
  </h5>
</DocumentFragment>
`;

exports[`FormHeading Snapshots themes FormHeading dark theme snapshot 1`] = `
<DocumentFragment>
  .c0 {
  color: hsl(0,0%,100%,.87);
  margin: 0;
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
  color: hsl(0,0%,100%,.87);
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  font-size: var(--bds-font-tag-h5-mobile-font-size);
  line-height: var(--bds-font-tag-h5-mobile-line-height);
}

.c0.align--left {
  text-align: left;
}

.c0.align--center {
  text-align: center;
}

.c0.align--right {
  text-align: right;
}

.c0.display--block {
  display: block;
}

.c0.display--inline {
  display: inline;
}

.c0 b,
.c0 bold,
.c0 strong {
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
}

.c0.bold {
  font-weight: 500;
}

@media only screen and (min-width:66.0625rem) {
  .c0 {
    font-size: var(--bds-font-tag-h5-desktop-font-size);
    line-height: var(--bds-font-tag-h5-desktop-line-height);
  }
}

<h5
    class="c0 typography variant--h5 align--left form-heading"
    color="text_primary"
  >
    Heading
  </h5>
</DocumentFragment>
`;

exports[`FormHeading Snapshots themes FormHeading light theme snapshot 1`] = `
<DocumentFragment>
  .c0 {
  color: hsl(0,0%,0%,.87);
  margin: 0;
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
  color: hsl(0,0%,0%,.87);
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  font-size: var(--bds-font-tag-h5-mobile-font-size);
  line-height: var(--bds-font-tag-h5-mobile-line-height);
}

.c0.align--left {
  text-align: left;
}

.c0.align--center {
  text-align: center;
}

.c0.align--right {
  text-align: right;
}

.c0.display--block {
  display: block;
}

.c0.display--inline {
  display: inline;
}

.c0 b,
.c0 bold,
.c0 strong {
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
}

.c0.bold {
  font-weight: 500;
}

@media only screen and (min-width:66.0625rem) {
  .c0 {
    font-size: var(--bds-font-tag-h5-desktop-font-size);
    line-height: var(--bds-font-tag-h5-desktop-line-height);
  }
}

<h5
    class="c0 typography variant--h5 align--left form-heading"
    color="text_primary"
  >
    Heading
  </h5>
</DocumentFragment>
`;
